src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -368,6 +374,31 @@ static MvcCoreLoggerExtensions()
                 LogLevel.Debug,
                 5,
                 "Current action does not explicitly specify any content types for the response.");
+
+            _multipleRangesNotSupported = LoggerMessage.Define(
+                LogLevel.Warning,
+                6,
+                "Multiple ranges is not supported.");
+
+            _ifMatchPreconditionFailed = LoggerMessage.Define<EntityTagHeaderValue>(
+                LogLevel.Debug,
+                7,
+                "Current request's If-Match header check failed as the file's current etag '{CurrentETag}' does not match with any of the supplied etags.");
```

Comment:
Maybe something like 

> Specified entity tag in the If-Match precondition does not match the calculated value '{CurrentETag}' for the resource.

Reply:
'calculated value' is not clear for me. (BTW there can be more than one etag in If-Match header)